core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4820,14 +4820,18 @@ public boolean isSubjectToMandatoryReadPermissionCheck(String restOfPath) {
         }
 
         // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-        if (restOfPath.matches("/computer/[^/]+/slave-agent[.]jnlp")
+        if ((isAgentJnlpPath(restOfPath, "jenkins") || (isAgentJnlpPath(restOfPath, "slave")))
             && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
             return false;
         }
 
         return true;
     }
 
+    private boolean isAgentJnlpPath(String restOfPath, String prefix) {
+        return restOfPath.matches("/computer/[^/]+/" + prefix+ "-agent[.]jnlp");
```

Comment:
Why are we backporting a cosmetic change that touches security-sensitive code? This feels like an unnecessary risk.

Reply:
Well, It is that risk vs. terminology fix postponed by 12 weeks. The change seems safe to me. Any other thoughts?

Reply:
seems fine to me to backport

Reply:
Not a great backport for the same reasons I listed in https://github.com/jenkinsci/jenkins/pull/5010#pullrequestreview-516747246

Reply:
> terminology fix postponed by 12 weeks

I think a terminology fix can wait.